services/headless-lms/server/src/controllers/main_frontend/course_instances.rs

```diff
@@ -443,6 +443,28 @@ async fn get_user_progress_for_course_instance(
     token.authorized_ok(web::Json(user_course_instance_progress))
 }
 
+/**
+ POST /api/v0/main-frontend/course-instance/:course_instance_id/generate-join-code - Generates a code that is used as a part of URL to join course
+*/
+#[instrument(skip(pool))]
+async fn generate_join_code_for_course_instance(
+    id: web::Path<Uuid>,
+    pool: web::Data<PgPool>,
+    user: AuthUser,
+) -> ControllerResult<HttpResponse> {
+    let mut conn = pool.acquire().await?;
+    let token = authorize(
+        &mut conn,
+        Act::Edit,
+        Some(user.id),
+        Res::CourseInstance(*id),
+    )
+    .await?;
+    let code = Uuid::new_v4().to_string();
```

Comment:
I think a random string would be better. Also, we have to keep in mind that these strings should be long enough that no one can guess them, and we have to keep them easy to type. 
You could for example generate a random string that excludes look-alike characters. (https://rust-lang-nursery.github.io/rust-cookbook/algorithms/randomness.html#create-random-passwords-from-a-set-of-user-defined-characters)